Doc/library/decimal.rst

```diff
@@ -1902,17 +1902,18 @@ to work with the :class:`Decimal` class::
        3.141592653589793238462643383
 
        """
-       getcontext().prec += 2  # extra digits for intermediate steps
-       three = Decimal(3)      # substitute "three=3.0" for regular floats
-       lasts, t, s, n, na, d, da = 0, three, 3, 1, 0, 0, 24
-       while s != lasts:
-           lasts = s
-           n, na = n+na, na+8
-           d, da = d+da, da+32
-           t = (t * n) / d
-           s += t
-       getcontext().prec -= 2
-       return +s               # unary plus applies the new precision
+       with localcontext() as ctx:
+           ctx.prec += 2  # extra digits for intermediate steps
```

Comment:
Could you leave these comments aligned with the one on line 1915?

Otherwise this looks good, thanks!

Reply:
Done!